scripts/src/report/verifier_report.py

```diff
@@ -204,27 +204,33 @@ def validate(report_path):
         has_kubeversion_outcome, _ = get_chart_testing_result(report_data)
         if has_kubeversion_outcome:
             chart = report_info.get_report_chart(report_path)
-            if KUBE_VERSION_ATTRIBUTE in chart:
-                kube_supported_ocp_versions_string = indexannotations.getOCPVersions(
-                    chart[KUBE_VERSION_ATTRIBUTE]
-                )
-                try:
-                    kube_supported_versions = semantic_version.NpmSpec(
-                        kube_supported_ocp_versions_string
-                    )
-                except ValueError:
-                    if v1_0_profile:
-                        return True, ""
-                    else:
-                        return (
-                            False,
-                            f"Kube Version {chart[KUBE_VERSION_ATTRIBUTE]} translates to an invalid OCP version range {kube_supported_ocp_versions_string}",
-                        )
-            else:
-                if v1_0_profile:
-                    return True, ""
-                else:
-                    return False, f"{KUBE_VERSION_ATTRIBUTE} missing from chart!"
+            kube_supported_ocp_versions_string = ocp_version_range
+            kube_supported_versions = semantic_version.NpmSpec(
+                kube_supported_ocp_versions_string
+            )
+            # TODO: no errors when profile v1 ?
+
+            # if KUBE_VERSION_ATTRIBUTE in chart:
+            #     kube_supported_ocp_versions_string = indexannotations.getOCPVersions(
+            #         chart[KUBE_VERSION_ATTRIBUTE]
+            #     )
+            #     try:
+            #         kube_supported_versions = semantic_version.NpmSpec(
+            #             kube_supported_ocp_versions_string
+            #         )
+            #     except ValueError:
+            #         if v1_0_profile:
+            #             return True, ""
+            #         else:
+            #             return (
+            #                 False,
+            #                 f"Kube Version {chart[KUBE_VERSION_ATTRIBUTE]} translates to an invalid OCP version range {kube_supported_ocp_versions_string}",
+            #             )
+            # else:
+            #     if v1_0_profile:
+            #         return True, ""
+            #     else:
+            #         return False, f"{KUBE_VERSION_ATTRIBUTE} missing from chart!"
```

Comment:
Think we can remove this instead of commenting?

Reply:
Are v1.0 profiles still supported ?

This piece of code contains logic to ignore errors when the Kube Version Range is not set in a report (for profile v1.0 only). Should I reimplement this logic ?

FYI currently I counted 30 reports that are using the v1.0 profile:
```
❯ for i in $(find charts/ -name report.yaml); do profile_version=$(podman run --rm -v "${PWD}":/workdir:z mikefarah/yq .metadata.tool.profile.version $i); if grep -q "1.0" <<< $profile_version; then echo $i $profile_version; fi; done
charts/community/redhat/dotnet/0.0.1/report.yaml v1.0
charts/community/redhat/wildfly/1.4.0/report.yaml v1.0
charts/community/redhat/wildfly/1.5.2/report.yaml v1.0
charts/partners/fortanix/dsm-secrets-injector-chart/2.0/report.yaml v1.0
charts/partners/fortanix/dsm-secrets-injector-chart/2.2-4.6/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.10/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.12/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.13/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.14/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.3/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.4/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.5/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.6/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.7/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.8/report.yaml v1.0
charts/partners/ibm/ibm-object-storage-plugin/2.1.9/report.yaml v1.0
charts/partners/ibm/ibm-oms-ent-prod/6.0.2/report.yaml v1.0
charts/partners/ibm/ibm-oms-pro-prod/6.0.2/report.yaml v1.0
charts/partners/ibm/ibm-operator-catalog-enablement/1.2.2/report.yaml v1.0
charts/partners/ibm/ibm-spectrum-protect-plus-prod/1.2.1/report.yaml v1.0
charts/partners/nirmata/kyverno/v2.1.2/report.yaml v1.0
charts/partners/test-org/psql-service/0.1.10/report.yaml v1.0
charts/redhat/redhat/dotnet/0.0.1/report.yaml v1.0
charts/redhat/redhat/eap-xp3/1.0.0/report.yaml v1.0
charts/redhat/redhat/eap74/1.1.0/report.yaml v1.0
charts/redhat/redhat/nodejs-ex-k/0.2.1/report.yaml v1.0
charts/redhat/redhat/nodejs/0.0.1/report.yaml v1.0
charts/redhat/redhat/psql-sed/1.0.0/report.yaml v1.0
charts/redhat/redhat/quarkus/0.0.3/report.yaml v1.0
charts/redhat/redhat/spring-boot-rest-http/0.0.1/report.yaml v1.0
```

Reply:
The issue is that (with my changes) the pipeline is likely to fail for reports with profile = 1.0, as the step which runs get-ocp-range will fail.

I believe I need to extract the profile version as part of the pipeline and handle the 1.0 case gracefully there (e.g. set a default value for the OCP_RANGE.

Reply:
The profile versioning seems to an incremental change to expectations imposed on a given chart. To that end, I would expect all new charts to come in at the latest profile.

I'm unsure of the motivation behind allowing backwards compatibility with the profile versions. I would be interested in removing this concept of profile versioning.

`get-ocp-range` needs the kubeVersion field, right? That's been required since profile version 1.0 as I understand it. What is it that will break here for these?

Reply:
get-ocp-range will fail if kubeVersion is missing.

Maybe the yq command will have failed also beforehand: https://github.com/openshift-helm-charts/development/pull/264/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R229

Reply:
I've implemented the following:

* added a step that extracts the profile version set in report.yaml
* added logic to fail the job if there is an error in get-ocp-range, only if the profile version is not 1.0

This is feature equivalent, as prior to this PR we were also attempting (in python) to calculate the OCP range and ignoring failures if the profile was 1.0.

On the down side it adds two extra steps to an already very long job.

> To that end, I would expect all new charts to come in at the latest profile.

If we decide to enforce this, we can simplify several parts of our codebase. I'd be in favor :+1: